src/components/HabitForm.js

```diff
@@ -1,72 +1,78 @@
-import { useEffect, useState } from 'react';
-
-import styled from 'styled-components/macro';
-
+import PropTypes from "prop-types";
+import { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+import { v4 as uuidv4 } from "uuid";
 
+import styled from "styled-components/macro";
 
 HabitForm.propTypes = {
-    headlineText: PropTypes.string,
-    onAddHabit: PropTypes.func,
-    häufigkeit: PropTypes.arrayOf(PropTypes.object),
-    habitToEdit: PropTypes.object,
-  };
-
+  headlineText: PropTypes.string,
+  onAddHabit: PropTypes.func,
+  häufigkeit: PropTypes.arrayOf(PropTypes.object),
+  setHabitToEdit: PropTypes.func,
+};
 
 export default function HabitForm({
-    headlineText,
-    onAddHabit,
-    häufigkeit,
-    habitToEdit,
-}) { 
-const initialState = ()=>{
-     return JSON.parse(window.localStorage.getItem('habit')) 
-};
+  headlineText,
+  onAddHabit,
+  onEditHabit,
+  häufigkeit,
+  setActivePage,
+  habitToEdit,
+  setHabitToEdit,
+  handleFormSubmit,
+  setShowsEditModal,
+}) {
+  const initialHabit = {
+    ziel: "",
+    häufigkeit: "",
+  };
 
-const [habit, setHabit] = useState(initialState());
-const [isError, setIsError] = useState(false);
-const [fieldName, setFieldName] = useState('');
-const [fieldValue, setFieldValue] = useState('');
-const [frequency, setFrequency] = useState('');
-const [frequencyName, setFrequencyName] = useState('');
+  const [habit, setHabit] = useState(habitToEdit ?? initialHabit);
+  const [isError, setIsError] = useState(false);
+  const [frequency, setFrequency] = useState("");
 
-function updateField(event) {
-    setFieldName(event.target.name);
-    setFieldValue(event.target.value);  
-}
+  const placeholderText = `neues Ziel tippen...
+`;
 
-function updateFrequency(event) {
-    setFrequencyName(event.target.name);
-    setFrequency(event.target.value);
-}
+  function updateHabit(event) {
+    const fieldName = event.target.name;
+    let fieldValue = event.target.value;
+    setHabit({ ...habit, [fieldName]: fieldValue });
+  }
 
-function handleFormSubmit(event) {
+  function handleFormSubmit(event) {
     event.preventDefault();
-    let habitArray = habit?[...habit]:[];
-    habitArray.push({[fieldName]: fieldValue, [frequencyName]: frequency })
-    setHabit(habitArray);
+    if (habitToEdit) {
+      onEditHabit(habit);
+      setShowsEditModal(false);
+    } else {
+      onAddHabit({ ...habit, id: uuidv4() });
+    }
   }
-   
-  useEffect(() => {
-    window.localStorage.setItem('habit', JSON.stringify(habit))
-  }, [habit]);
 
-  function addHabit(habit) {
-    setHabit([...habit, habit]);
-  }
+  let history = useHistory();
+  const goToPreviousPath = () => {
+    history.push("/today");
+  };
 
   return (
     <Form onSubmit={handleFormSubmit}>
       <label htmlFor="ZielName">Ziel</label>
       <Ziel
         type="text"
         name="ziel"
-        onChange={updateField}
-        value={fieldValue}
+        onChange={updateHabit}
```

Comment:
For slightly more specificity for future you, why not try this as `handleUpdateHabitChange`